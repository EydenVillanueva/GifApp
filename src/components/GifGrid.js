import React from 'react';
import PropTypes from 'prop-types';

import GifGridItem from "./GifGridItem";
import {useFetchGifs} from "../hooks/useFetchGifs";

const GifGrid = ({category}) => {
	
	const { loading, data: images } = useFetchGifs(category);
	
	const renderImages = (images) => {
		return <ol>
			{ images.map( ({id, title, url}) => <GifGridItem key={id} title={title} url={url} /> )}
		</ol>;
	}
	
	return (
		<div className="card-grid" >
			<h3> {category} </h3>
			{ loading && <p>'Loading...'</p> }
			{renderImages(images)}
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired
};

export default GifGrid;


