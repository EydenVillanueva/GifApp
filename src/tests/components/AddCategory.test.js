import React from 'react';
import { shallow } from "enzyme";

import { AddCategory } from "../../components/AddCategory";

describe('AddCategory component test suite', () => {
	
	const setCategories = jest.fn();
	let wrapper;
	
	beforeEach( () => {
		jest.clearAllMocks();
		wrapper = shallow( <AddCategory setCategories={setCategories}/> );
	});
	
	test('Should show correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
	
	test('Should change the text field', () => {
		const input = wrapper.find('input');
		const value = 'Hola mundo';
		
		input.simulate('change', { target: { value }});
		expect(wrapper.find('p').text()).toBe(value);
	});
	
	test('Should not post the information on submit', () => {
		wrapper
			.find('form')
			.simulate('submit', {preventDefault(){}});
		
		expect( setCategories ).not.toHaveBeenCalled();
	});
	
	test('Should call the setCategories and clean the text field', () => {
		const value = 'input changed';
		
		// simulate the handleInputChange and handleSubmit
		wrapper.find('input').simulate('change', { target: {value}});
		wrapper.find('form').simulate('submit', { preventDefault(){} });
		
		// setCategories should be called
		expect( setCategories ).toBeCalled();
		expect( setCategories ).toHaveBeenLastCalledWith( expect.any(Function) );
		// input value should be rested to ''
		expect(wrapper.find('input').props().value).toBe('');
	});
});