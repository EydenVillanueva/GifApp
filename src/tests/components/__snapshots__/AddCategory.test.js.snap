// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component test suite Should show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
